pkg/upload/path_provider.go

```diff
@@ -0,0 +1,117 @@
+package upload
+
+import (
+	"path"
+	"sync"
+	"time"
+
+	"github.com/rs/xid"
+)
+
+// PathProvider provides path by request to upload data. The provided path is built from '<prefix>/<partition>/<unique id>'.
+// The partition is updated every interval or when the number of IDs generated passes the specified size.
+type PathProvider struct {
+	mu  sync.Mutex
+	cfg *PathProviderConfig
+	// timestamp used to calculate the time based ID value
+	timestamp time.Time
+	// count make sure we will not produce more than 'count' IDs
+	count int
+	// partition current set and reset by size and time
+	partition string
+}
+
+const (
+	DefaultDataSize     = 50000
+	DefaultDataInterval = 5 * time.Minute
```

Comment:
I think this default interval is a bit small?
Maybe we should start with an hour?